src/mediaelement.js

```diff
@@ -65,6 +65,36 @@ export default class MediaElement extends WebAudio {
         this.mediaListeners.error = () => {
             this.fireEvent('error', 'Error loading media element');
         };
+        this.mediaListeners.abort = () => {
+            this.fireEvent('abort');
+        };
+        this.mediaListeners.emptied = () => {
+            this.fireEvent('emptied');
+        };
+        this.mediaListeners.canplaythrough = () => {
+            this.fireEvent('canplaythrough');
+        };
+        this.mediaListeners.loadeddata = () => {
+            this.fireEvent('loadeddata');
+        };
+        this.mediaListeners.loadedmetadata = () => {
+            this.fireEvent('loadedmetadata');
+        };
+        this.mediaListeners.loadstart = () => {
+            this.fireEvent('loadstart');
+        };
+        this.mediaListeners.ratechange = () => {
+            this.fireEvent('ratechange');
+        };
+        this.mediaListeners.stalled = () => {
+            this.fireEvent('stalled');
+        };
+        this.mediaListeners.suspend = () => {
+            this.fireEvent('suspend');
+        };
+        this.mediaListeners.timeupdate = () => {
+            this.fireEvent('timeupdate');
```

Comment:
@katspaugh yes I will add the relevant ones for example for `loadedmetadata` it can have the following data returned

- duration: The length of the media in seconds.
- videoWidth: The width of the video in pixels.
- videoHeight: The height of the video in pixels.
- audioTracks: An array of all available audio tracks for the media.

for `timeupdate` it's
- currentTime
- duration


Some of the events are relevant in webaudio as well but a bit different `onloadedmetadata` but yea they can be accessed via `wavesurfer.backend.media.addEventListener` instead of proxying. For some of these might be better to make it standard text that supports both backends depending on which one is active such as `wavesurfer.on('loadedmetadata')` or even `wavesurfer.on('metadata')`. Suggestions are welcomed let me know what you think.

Reply:
If you think we lack a metadata event atm, and users require it, sure, why not.